js/blocks/controls/fetch-input.js

```diff
@@ -152,13 +150,34 @@ class FetchInput extends Component {
 	 * That has its own handler, which will eventually hide the Popover.
 	 */
 	onBlur( event ) {
-		if ( event.relatedTarget && ! event.relatedTarget.classList.contains( 'bl-fetch-input__suggestion' ) ) {
+		if (
+			event.relatedTarget &&
+			! event.relatedTarget.classList.contains( 'components-popover__content' ) &&
+			! event.relatedTarget.classList.contains( 'bl-fetch-input__suggestion' )
+		) {
 			this.setState( {
 				showSuggestions: false,
 			} );
+
+			if ( '' === this.props.value ) {
+				return;
+			}
+
+			const matchingResults = this.state.results.filter(
+				suggestions => ( suggestions.slug === this.props.value )
+			);
+
+			if ( ! matchingResults.length ) {
+				this.selectLink( this.state.results[ this.state.selectedSuggestion ] );
+			}
```

Comment:
If the user blurs the field with only a partial name match, then populate the field with the full suggestion (otherwise we get an invalid user).

Reply:
Nice improvement here. That definitely makes more sense than leaving the `<input>` empty.